py2many/cli.py

```diff
@@ -394,7 +395,10 @@ def _process_dir(settings, source, outdir, env=None, _suppress_exceptions=True):
                 format_errors.append(path)
         except Exception as e:
             print(f"Error: Could not transpile: {path}")
-            print(f"Due to: {e.__class__.__name__} {e}")
+            if isinstance(e, AstNotImplementedError):
```

Comment:
```
 if hasattr(e, "lineno"): ...
 ```
 is more precise here. I can imagine we'll want to throw different types of exceptions from the transpiler

Reply:
It would need to be `if hasattr(e, "lineno") and hasattr(e, "col_offset"):` which is silly.  Those properties should be defined in an interface.

Currently the interface is `AstNotImplementedError`.  If we need others, we can create `exceptions.AstErrorBase` and `AstNotImplementedError` and others can subclass that.

Reply:
Sound good. Let's check `isinstance(AstErrorBase)` and have two subclasses `TypeError` and `AstNotImplementedError`.

Reply:
I'll merge this as-is to make forward progress. Let's work on the Exception hierarchy in stacked diffs.